realm/realm-library/src/main/cpp/jni_util/hack.cpp

```diff
@@ -72,6 +67,7 @@ static void check_memmove()
     if (ptr != array + 1) {
         Log::e("memmove is broken on this device. switch to the builtin implementation.");
         s_wrap_memmove_ptr = &hacked_memmove;
+        s_wrap_memcpy_ptr =  &hacked_memmove;
```

Comment:
For example, on FreeBSD and OS X `memmove` and `memcpy` are calling `bcopy`. The only difference is that `memmove` returns the new end. I mean, this are the naive implementation in C. FreeBSD (haven't check OS X) has optimized versions written in assembler for a number of architectures like Intel and ARM.